### DISCUSSION TITLE
API can not be set to private

### DISCUSSION THREAD

**Discussion by yjeanrenaud** [YOURLS/YOURLS#3103] [Q&A] [answered]
I have a proper running install of version 1.8.2.
I use `define( 'YOURLS_PRIVATE', true );` which works for admin, but the passwordless API is still available to public.
Also  `define('YOURLS_PRIVATE_API', true);` seems not to work.
Any ideas?

**Comment by dgw** [comment]
I don't understand the question. The API ultimately hands off to `includes/auth.php` if `YOURLS_PRIVATE` is set, and the auth code sends a 403 Forbidden error if the authentication is missing or invalid.

**Comment by ozh** [comment, marked as answer]
1. `YOURLS_PRIVATE` : set to `true` to make your YOURLS instance private, eg `sho.rt/admin` requires login/pwd
2. `YOURLS_PRIVATE_API` : **if** `YOURLS_PRIVATE` is `true`, set this constant to **`false`** to make the API available to the public *without* login/pwd.

So, yes, you can set the API to be private : just set the first of these 2 constants to `true`. Example : http://yourls.org/admin & http://yourls.org/yourls-api.php, a fine example of a private YOURLS where, by default, the API is also private.

**Reply by yjeanrenaud** [reply]
ah thank you. so it's really the token that serves as auth mechanism in private mode. I misunderstood that. I thought, setting yourls_private_api true means noone can access the API anymore except when a valid login cookie is set.
